cleverhans/model.py

```diff
@@ -0,0 +1,221 @@
+from abc import ABCMeta
+
+
+class Model(object):
+    """
+    An abstract interface for model wrappers that exposes model symbols
+    needed for making an attack. This abstraction removes the dependency on
+    any specific neural network package (e.g. Keras) from the core
+    code of CleverHans. It can also simplify exposing the hidden features of a
+    model when a specific package does not directly expose them. For an
+    example of why this is useful, see the paper "Adversarial Manipulation
+    of Deep Representations" (https://arxiv.org/abs/1511.05122).
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
```

Comment:
For the main class, I would prefer going with Ian's suggestion. Making design choices based on Keras has been a source of problem in the past for us. 

However, Nicholas is in the process of putting the model argument in the KerasModelWrapper's init method as an optional argument.